### PR TITLE
[mssql] Add new plugin

### PR DESCRIPTION
**Pull request by tanaka-takayoshi**
Adds a new plugin for Microsoft SQL Server on Linux.

Collect log files.
Collect journal logs.
Collect keytab file info if AD auth is configured.

Signed-off-by: Takayoshi Tanaka <tatanaka@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

File: sos/plugins/mssql.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        # errorlogfile = /var/opt/mssql/log
+        # [sqlagent]
+        # errorlogfile = /var/opt/mssql/log/sqlagentstartup.log
+        # [network]
+        # kerberoskeytabfile = /var/opt/mssql/secrets/mssql.keytab
+        # ```
+        # Default values are described in the document:
+        # https://docs.microsoft.com/en-us/sql/linux
+        # /sql-server-linux-configure-mssql-conf
+        # ?view=sql-server-linux-2017#mssqlconf-format
```
Comment (bmr-cymru) [inline sos/plugins/mssql.py]: Let's just drop this URL; interested parties can easily find it if they need it. It's just a plan INI file, so we can simply document the sections and keys that we care about.

File: sos/plugins/mssql.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                self.add_cmd_output('ls -l %s' % kerberoskeytabfile)
+                self.add_cmd_output('klist -e -k %s' % kerberoskeytabfile)
+            else:
+                self._log_error(keytab_err)
+
+        # Expecting mssql_conf doesn't includeno sensitive information.
+        self.add_copy_spec([
+            mssql_conf,
+            errorlogfile + '/*',
+            sqlagent_errorlogfile
```
Comment (bmr-cymru) [inline sos/plugins/mssql.py]: This should only add all the log files if `--all-logs` is given. Please also implement size limiting for the log files only (no need for configuration). Running `git grep 'get_option.*log_size'` will give you plenty of examples to look at.

File: sos/plugins/mssql.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                    if section == '[filelocation]':
+                        errorlogfile = words[1].strip()
+                    elif section == '[sqlagent]':
+                        sqlagent_errorlogfile = words[1].strip()
+                elif words[0].strip() == 'kerberoskeytabfile':
+                    if section == '[network]':
+                        kerberoskeytabfile = words[1].strip()
+        except IOError as ex:
+            self._log_error('Could not open conf file %s: %s' % (mssql_conf,
+                                                                 ex))
```
Comment (bmr-cymru) [inline sos/plugins/mssql.py]: This is a little more complex than we'd generally like for plugin config reading code, but the alternative would be bringing in something like `ConfigParser` which isn't ideal right now: with any luck we will be able to replace this with a generic config file reading facility in the near future.

File: sos/plugins/mssql.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+from sos.plugins import Plugin, RedHatPlugin
```
Comment (bmr-cymru) [inline sos/plugins/mssql.py]: Generally in `sos` we import the `sos` modules first and then general Python modules. This is not the usual Python convention but it's something we've been doing for a long time - sticking to it means that if we do decide to change in the future it is easier to apply a scripted change to all source files at once.

File: sos/plugins/mssql.py (hunk 3 above)
Comment (tanaka-takayoshi) [reply sos/plugins/mssql.py]: I pushed a new commit as per comment. However, should I fix this code right now?

Comment (bmr-cymru) [reply sos/plugins/mssql.py]: It is quite complex but I think it is OK for now: we do not yet have the generic config reading work done, so there is no better solution for now.

**Pull request closed without merging**